lib/src/models/transaction_uploader.dart

```diff
@@ -131,19 +131,23 @@ class TransactionUploader {
     //  throw StateError('Unable to validate chunk: $_chunkIndex');
 
     // Catch network errors and turn them into objects with status -1 and an error message.
-    final res = await _api.post('chunk', body: json.encode(chunk));
+    try {
+      final res = await _api.post('chunk', body: json.encode(chunk));
 
-    _lastRequestTimeEnd = DateTime.now().millisecondsSinceEpoch;
-    lastResponseStatus = res.statusCode;
+      _lastRequestTimeEnd = DateTime.now().millisecondsSinceEpoch;
+      lastResponseStatus = res.statusCode;
 
-    if (lastResponseStatus == 200) {
-      _chunkIndex++;
-    } else {
-      lastResponseError = getResponseError(res);
-      if (FATAL_CHUNK_UPLOAD_ERRORS.contains(lastResponseError)) {
-        throw StateError(
-            'Fatal error uploading chunk: $_chunkIndex: $lastResponseError');
+      if (lastResponseStatus == 200) {
+        _chunkIndex++;
+      } else {
+        lastResponseError = getResponseError(res);
+        if (FATAL_CHUNK_UPLOAD_ERRORS.contains(lastResponseError)) {
+          throw StateError(
```

Comment:
This is now broken. We won't ever fail on fatal error until we hit 100 errors and even when we do the error message will not be this. It'll be whatever is in `$lastResponseStatus` and  `$lastResponseError`.

Reply:
Should I remove this part? Since we never get back any fatal errors?

Reply:
I don't think so. I believe it's a hold over from previous gateway and/or node behaviors. try catches should be very tight and focused. If you're interested in catching the post error, then catch and handle the post error with a specific message/behavior. If you're also interested in catching the call to getResponseError, then wrap that individually and log/behave appropriately. Large try/catches are a code smell.